src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -245,16 +247,23 @@ public static async Task<JsonDocument> GetJsonCache(HttpClient client, string ur
                         AllowTrailingCommas = true,
                     });
                     break;
-                default: 
-                    doc = await JsonDocument.ParseAsync(await result.Content.ReadAsStreamAsync());
+                default:
+                    Stream responseStream = await result.Content.ReadAsStreamAsync();
+                    contentLength = responseStream.Length;
```

Comment:
Huh - well that's pretty convenient. I thought we might need a stream-wrapper approach in order to dynamically calculate the total length of the stream, but it looks like we don't have to do that (at least for now).